src/Modules/UngradedModules/FreeModuleHom.jl

```diff
@@ -553,9 +559,11 @@ Homogeneous module homomorphism)
 ```
 """
 function image(h::FreeModuleHom)
-  si = [x for x = map(h, basis(domain(h))) if !iszero(x)]
+  si = filter(!iszero, copy(images_of_generators(h)))
```

Comment:
Why the `copy`? The `filter` function returns a new list anyway?
```suggestion
  si = filter(!iszero, images_of_generators(h))
```